orttraining/orttraining/python/training/onnxblock/optim/optim.py

```diff
@@ -125,77 +60,231 @@ def build(self, gradients_name: str):
-    """Builds AdamW optimizer onnxblock for the given training parameters.
+class _OptimizerBase(blocks.Block):
+    def __init__(self):
+        super().__init__()
 
-    Creates a block that updates the model parameters based on the calculated
-    gradient following the AdamW algorithm.
+    def build_optimizer_node(
+        self,
+        input_names: List[str],
+        output_name: str,
+        node_name: str,
+        node_domain: str,
+        node_attributes: Dict,
+    ) -> str:
+        """
+        Build and append an optimizer node to the ONNX graph.
+
+        Args:
+            input_names (list): List of input tensor names for the optimizer node.
+            output_name (str): Output tensor name of the optimizer node.
+            node_name (str): Name of the optimizer node.
+            node_domain (str): Domain of the optimizer node.
+            node_attributes (dict): Additional attributes for the optimizer node.
+
+        Returns:
+            str: The output tensor name of the optimizer node.
+        """
+        onnx_model = self.base
+        optimizer_input_names = input_names
+        optimizer_output_names = [output_name]
+
+        # add the optimizer node to the onnx model
+        optimizer_node = onnx.helper.make_node(
+            node_name,
+            optimizer_input_names,
+            optimizer_output_names,
+            name=_graph_utils.generate_graph_name(node_name),
+            domain=node_domain,
+            **node_attributes,
+        )
 
-    Args:
-        bias_correction: bool indicating whether to perform bias correction.
-        betas: AdamW decay rate hyperparameters.
-        eps: term added to the denominator for computing the moments.
-        weight_decay: AdamW weight decay
-        clip_grad (optional): an instance of the ClipGradNorm. If not provided,
-                              gradient clipping will not be done.
+        onnx_model.graph.node.append(optimizer_node)
+
+        return output_name
 
-    Returns:
-        Returns a string of the output names from this optimizer node.
-    """
 
+class SGDOptimizer(_OptimizerBase):
+    def __init__(self):
+        super().__init__()
+
+    def build(
+        self,
+        learning_rate_name: str,
+        gradients_name: str,
+        params_name: str,
+    ) -> str:
+        """
+        Build an SGD optimizer node.
+
+        Args:
+            learning_rate_name (str): Name of the learning rate input tensor.
+            gradients_name (str): Name of the gradients input tensor.
+            params_name (str): Name of the weights input tensor.
+
+        Returns:
+            str: The output tensor name of the SGD optimizer node.
+        """
+
+        input_names = [learning_rate_name, gradients_name, params_name]
+
+        return self.build_optimizer_node(
+            input_names,
+            _graph_utils.generate_graph_name("update_completed"),
+            "SGDOptimizerV2",
+            "com.microsoft",
```

Comment:
domain name need not be part of the input arguments to the function since it is common and probably all future optimizers will share the same domain.